src/Npgsql/NpgsqlConnectionStringBuilder.cs

```diff
@@ -683,6 +683,73 @@ public ChannelBinding ChannelBinding
     }
     ChannelBinding _channelBinding;
 
+    /// <summary>
+    /// Controls the available authentication methods.
+    /// </summary>
+    [Category("Security")]
+    [Description("Controls the available authentication methods.")]
+    [DisplayName("Require Auth")]
+    [NpgsqlConnectionStringProperty]
+    public string? RequireAuth
+    {
+        get => _requireAuth;
+        set
+        {
+            AllowedAuthModes = ParseAuthMode(value);
+            _requireAuth = value;
+            SetValue(nameof(RequireAuth), value);
+        }
+    }
+    string? _requireAuth;
+
+    internal RequireAuthMode AllowedAuthModes { get; private set; }
```

Comment:
Should we keep the same naming here of the user-facing property instead of having two completely different names here (RequireAuth and AllowedAuthModes)?

Reply:
Renamed to `RequireAuthModes` which should align a bit better I guess?